### PR TITLE
feat(tasks): add FileGrepTask for server-side file grepping

### PR DESCRIPTION
**Pull request by sroussey**
- Introduced FileGrepTask to enable grepping of local files in Node.js and Bun environments.
- Updated browser, electron, and node task registration to include FileGrepTask.
- Enhanced task registration to return both FileGrepTask and FileLoaderTask.
- Added comprehensive tests for FileGrepTask functionality, including handling of local file paths and streaming CRLF files.
- Implemented input validation and error handling for various edge cases in the grepping process.

### CODE REVIEW

File: packages/tasks/src/task/FileGrepTask.ts [hunk 1]
```diff
@@ -0,0 +1,522 @@
...
+  readonly line: number;
+  readonly text: string;
+}
+
+/**
+ * Detects regex patterns prone to catastrophic backtracking (ReDoS).
+ * Checks for nested quantifiers like (a+)+, (a*)+, (a+)*, etc.
+ */
+function hasNestedQuantifiers(pattern: string): boolean {
+  const withoutClasses = pattern.replace(/\[(?:[^\]\\]|\\.)*\]/g, "X");
```
[the remark this answers is not in the document]
Comment (sroussey) [reply packages/tasks/src/task/FileGrepTask.ts]: The alert is real. `pattern.replace(/\[(?:[^\]\\]|\\.)*\]/g, "X")` rescans to the end from every unterminated `[`, so a pattern of N of them costs O(N²). Measured: 200k brackets takes ~3 s.

It is bounded here only because `assertSafeRegexPattern` runs the `regexMaxBracketCount` (100) check before calling this — a call-ordering invariant nothing enforces, and `hasUnsafeRegexShape` is exported without it.

Fixed at the root in #833 (which relocates this function to `packages/tasks/src/util/RegexSafety.ts` and would otherwise inherit the alert), as commit e3a3c99. A left-to-right scan stops at the first unterminated `[`: that proves no reachable `]` remains, and a later `[` sits at the same escape alignment so it cannot close either. Same input now screens in ~2 ms.

Two behavior changes fell out of reading escapes properly, both verified over 500k fuzzed patterns as never turning an unsafe verdict into a safe one:

- `\[` now reads as the literal it is. The regex matched it as a class opener, so `\[(a+)+]` stripped to `\X` and the nested quantifier inside was reported **safe**.
- each distinct class gets its own placeholder keyed by class text. A single shared `X` made `([a-z]|[A-Z])*` reduce to `(X|X)*`, so the alternation rule rejected a disjoint pattern that measures flat from n=22 to n=28. Identical classes still collapse together and are still caught.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: packages/tasks/src/task/FileGrepTask.ts [hunk 2]
```diff
@@ -0,0 +1,703 @@
...
+const inputSchema = {
+  type: "object",
+  properties: {
+    url: {
+      type: "string",
+      title: "URL",
+      description:
+        "URL to search (http://, https://). The Node/Electron build additionally accepts a `file://` URL or an absolute filesystem path, subject to the `filesystem:read` entitlement and any configured `roots`.",
+      format: "uri",
+    },
```
[the remark this answers is not in the document]
Comment (sroussey) [reply packages/tasks/src/task/FileGrepTask.ts:32]: The premise doesn't hold: nothing validates a value against `format`. `SchemaValidation.ts` only runs `validateFormatAnnotations`, which checks that the format *string itself* matches `FORMAT_PATTERN` — it never tests the instance. So `/tmp/x.txt` is not rejected anywhere, and the server tests exercising absolute paths and `file://` URLs pass for that reason rather than by accident.

`format` here is an annotation, carrying dataflow-compatibility and UI-hint meaning (the same slot `format: "model"` / `"knowledge-base"` / `"storage:tabular"` use), not a constraint.

The real residue is the one you're pointing at underneath: `url` is overloaded to mean URL-or-path, so the annotation is honest for the cross-platform build and misleading for the server build. That is tracked as #825 — a distinct `path` input port — which was kept out of #833 deliberately: it breaks the input schema for every existing workflow and needs a UI change, so it wants its own review rather than riding along in a security fix.

`anyOf` wouldn't help much in the meantime, since there is no validation to relax; it would only restate the ambiguity in the annotation. The `url` description was corrected in #833 to state the http(s)-vs-filesystem split explicitly, which is the part a UI or an authoring model actually reads today.

---
_Generated by [Claude Code](https://claude.ai/code)_
